lib/Doctrine/Migrations/MigrationTableUpdater.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Table;
+use Throwable;
+use function in_array;
+
+class MigrationTableUpdater
+{
+    /** @var Connection */
+    private $connection;
+
+    /** @var AbstractSchemaManager */
+    private $schemaManager;
+
+    /** @var MigrationTable */
+    private $migrationTable;
+
+    /** @var AbstractPlatform */
+    private $platform;
+
+    public function __construct(
+        Connection $connection,
+        AbstractSchemaManager $schemaManager,
+        MigrationTable $migrationTable,
+        AbstractPlatform $platform
+    ) {
+        $this->connection     = $connection;
+        $this->schemaManager  = $schemaManager;
+        $this->migrationTable = $migrationTable;
+        $this->platform       = $platform;
+    }
+
+    public function updateMigrationTable() : void
+    {
+        $fromTable = $this->getFromTable();
+        $toTable   = $this->migrationTable->getDBALTable();
+
+        $fromSchema = $this->createSchema([$fromTable]);
+        $toSchema   = $this->createSchema([$toTable]);
+
+        $queries = $fromSchema->getMigrateToSql($toSchema, $this->platform);
+
+        $this->connection->beginTransaction();
```

Comment:
this should be done only if the platform supports transational DDL IMO (which DBAL knows)

Reply:
DBAL doesn't know this currently. It only has the supportsTransactions() method. We discussed submitting an enhancement to the DBAL for this though. For now though, we figured having this here for everyone wouldn't hurt. Thoughts?

Reply:
ah indeed, I though it was something known by DBAL and used by doctrine/migrations, but doctrine/migrations always wraps the migration in a transaction, even if it would not be able to rollback it fully due to non transactional DDL (and migrations can ask to disable the usage of the transaction).

So yeah, keep it like that.